### ISSUE TITLE
[inductor][cpu] Accuracy failure on bmm max_autotune for offset input weights

### ISSUE THREAD

**Issue by frost-intel** [pytorch/pytorch#143770]
### 🐛 Describe the bug

Accuracy error is occurring for BMM max_autotune code when input weights have an offset. Issue is not reproducible on main due to #143102 but after #143141 lands, this issue shows up. Found testing torchbench `sam` model with `--amp`.

Here is a sample test to reproduce (could be added to `test/inductor/test_cpu_select_algorithm.py`):
```python
    @patches
    @torch.no_grad
    @unittest.skipIf(not TEST_MKL, "Test requires MKL")
    @dtypes(torch.bfloat16)
    def test_bmm_5d(self, dtype):
        class M(torch.nn.Module):
            def __init__(self):
                super().__init__()

            def forward(self, x, w):
                return x @ w[2]

        counters.clear()
        x = torch.randn(400, 196, 196).to(dtype=dtype)
        w = torch.randn(3, 400, 196, 80).to(dtype=dtype)
        mod = M().to(dtype=dtype).eval()
        with verify(dtype) as (atol, rtol):
            self.common(mod, (x, w), atol=atol, rtol=rtol)
        self.assertEqual(counters["inductor"]["select_algorithm_autotune"], 1)
```

The error seems to be related to taking the `as_strided` tensor in `normalize_shapes` in `cpp_gemm_template.py`, but more investigation is needed.

### Versions

Seen in main after cherry-picking from #143141

**Comment by frost-intel** [comment]
@leslie-fang-intel

**Comment by leslie-fang-intel** [comment]
Hi @frost-intel, since this issue is bmm related and depends on your PR https://github.com/pytorch/pytorch/pull/143141, maybe you can help to take a look when back from holiday, thanks.
